### ISSUE TITLE
进入死循环

### ISSUE THREAD

**Issue by jaceqin** [PaddlePaddle/PaddleSeg#3283]
### 问题确认 Search before asking

- [X] 我已经搜索过问题，但是没有找到解答。I have searched the question and found no related answer.

### 请提出你的问题 Please ask your question

![image](https://github.com/PaddlePaddle/PaddleSeg/assets/108056546/5288cd65-2249-4af6-a925-30e047692ca9)
代码运行到这里进入到了while的死循环，其中的值如下：
![image](https://github.com/PaddlePaddle/PaddleSeg/assets/108056546/7e55d508-27df-49e0-a9c4-871d93b789cc)
配置文件如下：
batch_size: 4
iters: 1000

train_dataset:
  type: OpticDiscSeg
  dataset_root: data/BraTs2020
  # train_path: data/BraTs2020/train_list.txt
  # num_classes: 2
  mode: train
  transforms:
    - type: ResizeStepScaling
      min_scale_factor: 0.5
      max_scale_factor: 2.0
      scale_step_size: 0.25
    - type: RandomPaddingCrop
      crop_size: [512, 512]
    - type: RandomHorizontalFlip
    - type: RandomDistort
      brightness_range: 0.5
      contrast_range: 0.5
      saturation_range: 0.5
    - type: Normalize

val_dataset:
  type: OpticDiscSeg
  dataset_root: data/BraTs2020
  # val_path: data/BraTs2020/val_list.txt
  # num_classes: 2
  mode: val
  transforms:
    - type: Normalize

optimizer:
  type: SGD
  momentum: 0.9
  weight_decay: 4.0e-5

lr_scheduler:
  type: PolynomialDecay
  learning_rate: 0.01
  end_lr: 0
  power: 0.9

loss:
  types:
    - type: CrossEntropyLoss
  coef: [1]

model:
  type: SegNet
  pretrained: Null

**Comment by jaceqin** [comment]
![image](https://github.com/PaddlePaddle/PaddleSeg/assets/108056546/8687c63b-271d-4779-9a68-c27e5ef0f993)
代码就运行图中1234，然后又循环1234，

**Comment by zengshao0622** [comment]
建议检查一下dataloader是否正常

**Comment by jaceqin** [comment]
![image](https://github.com/PaddlePaddle/PaddleSeg/assets/108056546/9a2b2324-c783-42f8-98d2-44c64342bfff)
代码循环运行上图中的1，2，3

**Comment by jaceqin** [comment]
![image](https://github.com/PaddlePaddle/PaddleSeg/assets/108056546/210879d0-1903-4d71-b3b0-50ddb6bc8a5c)
loader的值如上图

**Comment by jaceqin** [comment]
> 建议检查一下dataloader是否正常

dataloader正常

**Comment by Bobholamovic** [comment]
> > 建议检查一下dataloader是否正常
> 
> dataloader正常

建议执行如下语句检查`loader`的长度：
```print(sum(1 for _ in loader))```
